File: simple-service/src/test/java/com/example/demo/service/SparkServiceTests.java
package com.example.demo.service;

import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import java.util.List;

/**
 * Created by ksb on 2018. 2. 5..
 */
@RunWith(SpringRunner.class)
@SpringBootTest
public class SparkServiceTests {

    @Autowired
    private JavaSparkContext sc;

    @Autowired
    private SparkSession sparkSession;

    @Test
    public void TEST1() {
        JavaRDD<String> inFile = sc.textFile("../spark/csv/simple.csv");

        List<String> stringList = inFile.collect();

        System.out.println();
    }

    @Test
    public void TEST2() {
        Dataset<Row> df = sparkSession.read().json("../spark/resources/people.json");

        df.show();

//        df.printSchema();sunt().show();

        df.createOrReplaceTempView("people");
        Dataset<Row> sqlDF = sparkSession.sql("SELECT * FROM people");
        sqlDF.show();

        System.out.println();

        sparkSession.stop();
    }

    @Test
    public void TEST3() {
        Dataset<Row> df = sparkSession.read().json("../spark/resources/people.json");

        df.show();

        df.write().parquet("../spark/storage/parquet");

        System.out.println();

        sparkSession.stop();
    }

    @Test
    public void TEST4() {
        Dataset<Row> df = sparkSession.read().parquet("../spark/storage/parquet/part-00000-f573523a-a4c6-4757-8856-44b9951d08ba-c000.snappy.parquet");

        df.show();

        System.out.println();

        sparkSession.stop();
    }

}

File: spark-zip-code-mapping/src/test/scala/com/example/demo/service/SparkServiceTests.scala
package com.example.demo.service

import org.apache.spark.sql.{SparkSession}
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit4.SpringRunner

@RunWith(classOf[SpringRunner])
@SpringBootTest
class SparkServiceTests {

    var sparkSession : SparkSession = _

    @Autowired
    def setSparkSession(sparkSession : SparkSession) = this.sparkSession = sparkSession

    @Test
    def TEST0() = {
        val test1 = 1 to 10 mkString("[", ",", "]")
        val test2 = 1 until 10 mkString("[", ",", "]")
        val test3 = 1 to 10 by 3 mkString("[", ",", "]")
        val test4 = 1 until 10 by 3 mkString("[", ",", "]")

        println("test1 : %s", test1)
        println("test2 : %s", test2)
        println("test3 : %s", test3)
        println("test4 : %s", test4)
    }

    @Test
    def TEST1() : Unit = {
        val df = sparkSession.read.json("../spark/resources/people.json")

        df.show();

        println("TEST")
    }

    @Test
    def TEST2() : Unit = {
        val df = sparkSession.read.text("../spark/test.txt")

        df.show();

        println("TEST")
    }

}
